src/Fixer/ReturnNotation/ReturnAssignmentFixer.php

```diff
@@ -354,4 +373,98 @@ private function clearIfSave(Tokens $tokens, int $index): void
 
         $tokens->clearTokenAndMergeSurroundingWhitespace($index);
     }
+
+    /**
+     * @param int $index open brace index
+     *
+     * @return null|int index of the first token of a definition (lambda, anonymous class or match) or `null` if not an anonymous
+     */
+    private function isCloseBracePartOfDefinition(Tokens $tokens, int $index): ?int
```

Comment:
Not very important, but those `is*()` methods suggest they return `bool` but they actually return `?int`. I would rename them, e.g. for this one `getDefinitionFirstToken()`.

Reply:
Thanks for suggesting this change.
For now I'll as-is. If this PR gets closer to completion I might rename if I can think up better naming.